### PR TITLE
Add custom response support

### PR DESCRIPTION
**Pull request by ronnyhaase**
As I requested in #9

### CODE REVIEW

**Comment by ronnyhaase** [comment]
I also considered custom HTTP code but thought anything else than 401 doesn't make sense, but it would be easy to add.
I'm not sure If I like the tests, it's a lot of repetition for that minor feature, maybe someone has a better idea?

File: index.js [hunk 1]
```diff
@@ -2,33 +2,37 @@
...
+
     if (!secret) {
         throw Error('micro-jwt-auth must be initialized passing a secret to decode incoming JWT token')
     }
-    
+
+    if (!Array.isArray(whitelist)) {
+        config = whitelist || {}
+    }
+
```
Comment (ronnyhaase) [inline index.js]: I thought it's meaningful to turn `whitelist` to `config` so users don't have to pass an empty array if they want to pass a config but no whitelist.

File: test/index.test.js [hunk 2]
```diff
@@ -89,4 +89,44 @@ test('no need authorization bearer if whitelisted path', () => {
...
+    url: 'https://api.cabq.gov/domain/resources/1'
+  }
+
+  const response = {
+    writeHead: jest.fn().mockImplementation(),
+    end: jest.fn().mockImplementation()
+  };
+
+  const customRes = Math.random();
+  const result = jwtAuth('mySecret', [], { resAuthInvalid: customRes })(() => {})(request, response)
```
Comment (ronnyhaase) [inline test/index.test.js]: With empty whitelist

File: test/index.test.js [hunk 3]
```diff
@@ -89,4 +89,44 @@ test('no need authorization bearer if whitelisted path', () => {
...
+    url: 'https://api.cabq.gov/domain/resources/1'
+  }
+
+  const response = {
+    writeHead: jest.fn().mockImplementation(),
+    end: jest.fn().mockImplementation()
+  };
+
+  const customRes = Math.random()
+  const result = jwtAuth('mySecret', { resAuthMissing: customRes })()(request, response)
```
Comment (ronnyhaase) [inline test/index.test.js]: Skipping whitelist

File: index.js (hunk 1 above)
Comment (kandros) [reply index.js]: yeah, I think whis is a good idea, I felt it coming when first introduced the whitelist parameter, but didn't wanted to overdue back then, i thinkg now it makes way more sense to have an options object

File: test/index.test.js [hunk 4]
```diff
@@ -89,4 +89,44 @@ test('no need authorization bearer if whitelisted path', () => {
...
+    url: 'https://api.cabq.gov/domain/resources/1'
+  }
+
+  const response = {
+    writeHead: jest.fn().mockImplementation(),
+    end: jest.fn().mockImplementation()
+  }
+
+  const customRes = `${Math.random()}`
+  const result = jwtAuth('mySecret', [], { resAuthInvalid: customRes })(() => {})(request, response)
```
Comment (ronnyhaase) [inline test/index.test.js:156]: With (empty) whitelist

File: test/index.test.js [hunk 5]
```diff
@@ -89,4 +89,44 @@ test('no need authorization bearer if whitelisted path', () => {
...
+    url: 'https://api.cabq.gov/domain/resources/1'
+  }
+
+  const response = {
+    writeHead: jest.fn().mockImplementation(),
+    end: jest.fn().mockImplementation()
+  }
+
+  const customRes = `${Math.random()}`
+  const result = jwtAuth('mySecret', { resAuthMissing: customRes })()(request, response)
```
Comment (ronnyhaase) [inline test/index.test.js:175]: Skipping whitelist

**Review by kandros — CHANGES_REQUESTED** [review]
This branch should be rebased on the version currently in master.

Please also update the readme to document this new feature.

Thanks :)

**Comment by ronnyhaase** [comment]
Done

**Comment by kandros** [comment]
Hi Ronny, thank you for the help and sorry for they delay.

Updated version is now on npm

**Pull request merged**